admin/common_auth/views.py

```diff
@@ -24,3 +25,31 @@ def login(request):
 def logout(request):
     logout_user(request)
     return redirect('auth:login')
+
+def register(request):
+    # Creates User, takes selected permissions and assigns it to user, then saves resetform for email invitation
+    if request.user.is_authenticated():
+        return redirect('auth:login')
+    if request.method == 'POST':
+        form = CustomUserRegistrationForm(request.POST)
+        if form.is_valid():
+            email = form.cleaned_data['email']
+            first_name = form.cleaned_data['first_name']
+            last_name = form.cleaned_data['last_name']
+            password = form.cleaned_data['password']
+            user = MyUser.objects.create_user(username=email,
```

Comment:
You `MyUserManager.create_user` doesn't take those kwargs. You could change it to accept them, depending on other usages, or you could do this:
```python
        if form.is_valid():
            email = form.cleaned_data['email']
            first_name = form.cleaned_data['first_name']
            last_name = form.cleaned_data['last_name']
            password = form.cleaned_data['password1']
            user = MyUser.objects.create_user(email=email, password=password)
            user.first_name = first_name
            user.last_name = last_name
            user.save()
```